universal-application-tool-0.0.1/app/services/applicant/predicate/PredicateValue.java

```diff
@@ -0,0 +1,27 @@
+package services.applicant.predicate;
+
+import com.google.auto.value.AutoValue;
+
+/**
+ * Represents the value on the right side of a JsonPath (https://github.com/json-path/JsonPath)
+ * predicate expression. This value is usually a defined constant, such as a number, string, or
+ * array.
+ */
+@AutoValue
+public abstract class PredicateValue {
+
+  public static PredicateValue of(long value) {
+    return create(String.valueOf(value));
+  }
+
+  public static PredicateValue of(String value) {
+    // Escape the string value
+    return create("'" + value + "'");
```

Comment:
question (blocking): does it need to be wrapped in double instead of single quotes to work with JsonPath? strictly speaking JSON requires double quotes for string separators.

Reply:
Ah good catch - it technically works with single quotes in the [predicate tester](http://jsonpath.herokuapp.com/) but I will change to use double quotes since that is more standard for this library

Reply:
Done